ecodev_core/db_retrieval.py

```diff
@@ -190,3 +192,30 @@ def fields_order(query):
         return query.order_by(*[field.field for field in fields])
 
     return fields_order
+
+
+def get_sfield_columns(db_model: SQLModelMetaclass):
+    """
+    get all the columns flagged as sfields from schema
+    Args:
+        db_model (SQLModelMetaclass): db_model
+    """
+    return [
+        x.name
+        for x in inspect(db_model).c
+        if x.info.get(FILTER_ON) is True
+    ]
+
+
+def get_sfield_values(row: dict | SQLModelMetaclass,
+                      db_schema: SQLModelMetaclass | None = None) -> Dict:
```

Comment:
```suggestion
def get_sfield_values(row: dict | SQLModelMetaclass,
                      db_schema: SQLModelMetaclass | None = None
                      ) -> dict:
```

PI :). We  try to get rid of List and Dict from typing (already accessible by default now)

Reply:
Plus the name suggest that we get back a list, not a dict? Maybe just get_sfields?